include/psa/crypto_extra.h

```diff
@@ -1901,6 +1901,342 @@ static inline struct psa_pake_operation_s psa_pake_operation_init( void )
+ *
+ */
+struct psa_sign_interruptible_context_s
+{
+    size_t MBEDTLS_PRIVATE(num_ops);
+};
+
+typedef struct psa_sign_interruptible_context_s psa_sign_interruptible_context;
+
+/**
+ * \brief                       Set the maximum number of basic operations done
+ *                              in a row.
+ *
+ * \note                        Although the interpretation of this value is
+ *                              implicitly implementation defined, this value
+ *                              needed a default, thus this value defaults to
+ *                              1, which means the absolute minimum of work
+ *                              will be done each interruptible call.
+ *
+ * \note                        If more operations are needed to complete a
+ *                              computation, #PSA_ERROR_IN_PROGRESS will be
+ *                              returned by the function performing the
+ *                              computation. It is then the caller's
+ *                              responsibility to either call again with the
+ *                              same parameters until it returns 0 or an error
+ *                              code; or to free the relevant context if the
+ *                              operation is to be aborted.
+ *
+ * \note                        It is strictly required that all input
+ *                              parameters and the restart context be the same
+ *                              on successive calls for the same operation,
+ *                              but output parameters need not be the same;
+ *                              they must not be used until the function
+ *                              finally returns 0.
+ *
+ *                              This only applies to functions whose
+ *                              documentation mentions they may return
+ *                              #PSA_ERROR_IN_PROGRESS.
+ *
+ * \param max_ops               The maximum number of basic operations to
+ *                              be done in a row. Passing zero here has the
+ *                              same effect as passing the maximum value - i.e
+ *                              the operation will be done in one go. Lower
+ *                              (non-zero) values mean signature functions
+ *                              will block for a lesser maximum amount of time
+ *                              and conversely larger values will mean
+ *                              blocking for a larger maximum amount of time.
+ *                              The effect of changing this value will depend
+ *                              on the key used, the curve used and the
+ *                              underlying implementation, so we provide /c
+ *                              psa_sign_interruptible_get_num_ops() in order
+ *                              to help with tuning this value.
+ *
+ */
+void psa_sign_interruptible_set_max_ops( size_t max_ops );
```

Comment:
I don't think a global setting just for signature — or just for signature and verification — makes sense. If there's a global setting, it should encompass all asymmetric operations. Signature is what we want most urgently, but we'll definitely implement key agreement as well, and the PSA API will also define it for encryption/decryption even if we're not planning to implement that in Mbed TLS. So, assuming there is a global function, its name shouldn't include “sign”.